File: storev3/store-manager-web/src/main/webapp/js/service/TbSpecificationService.js
/*TbSpecificationService 服务层js offset, int limit*/
app.service("tbSpecificationService",function ($http) {
    this.findAll =function () {
      return $http.get('../tbSpecification/findAll.do');
    };


    this.findPage=function (page,size) {
        return $http.get('../tbSpecification/findPage.do?offset='+page+'&limit='+size);
    };

    this.add=function (entity) {
        return $http.post('../tbSpecification/add.do',entity);
    };

    this.update=function (entity) {
        return $http.post('../tbSpecification/update.do',entity);
    };

    this.findOne =function (id) {
        return $http.get('../tbSpecification/selectOne.do?id='+id);
    };

    this.dele=function (selectIds) {
        return $http.get('../tbSpecification/delete.do?ids='+selectIds);
    };


    this.search=function (page,size,searchEntity) {
        return $http.post('../tbSpecification/search.do?offset='+page +'&limit='+size,searchEntity);
    };


    this.getSpecificationList=function () {
        alert("进入service");
        return $http.post('../tbSpecification/getSpecificationList.do');
    }
});

File: storev3/store-sellergoods-interface/src/main/java/online/store/sellergoods/service/TbSpecificationService.java
package online.store.sellergoods.service;

import entity.PageModel;
import online.store.pojo.TbSpecification;
import online.store.pojoGroup.Specification;

import java.util.List;
import java.util.Map;


/**
 * (TbSpecification)表服务接口
 *
 * @author makejava
 * @since 2018-10-09 11:19:53
 */
public interface TbSpecificationService {

    /**
     * 通过ID查询单条数据
     *
     * @param id 主键
     * @return 实例对象
     */
    Specification selectByPrimaryKey(Long id);

    /**
     * 查询多条数据
     *
     * @param offset 查询起始位置
     * @param limit 查询条数
     * @return 对象列表
     */
    PageModel<TbSpecification> queryAllByLimit(int offset, int limit);


     /**
     * 查询所有数据
     *
     * @return 对象列表
     */
    List<TbSpecification> queryAllByLimit();


    /**
     * 新增数据
     *
     * @param specification 实例对象
     * @return 实例对象
     */
    TbSpecification insert(Specification specification);

    /**
     * 修改数据
     *
     * @param specification 实例对象
     * @return 实例对象
     */
    TbSpecification update(Specification specification);

    /**
     * 通过主键删除数据
     *
     * @param id 主键
     * @return 是否成功
     */
    boolean deleteByPrimaryKey(Long id);

    /**
     * 模糊查询并分页
     *
     * @param  tbSpecification 实例对象
     * @return pageModel
     */
    PageModel<TbSpecification> queryAllByLimit(TbSpecification tbSpecification, int offset, int limit);
    
    /**
     * 通过主键列表删除数据
     *
     * @param ids 主键列表
     * @return 是否成功
     */
     boolean deleteByPrimaryKeys(Long[] ids);

    List<Map> getSpecificationList();
}